add segmentFromTo and splitAt methods to gsBSpline class, and add a b…

jiyess opened the pull request:
I've frequently utilized the [merge](https://github.com/gismo/gismo/blob/a979dacec00b5fafb9ccea03e4a75b2682085982/src/gsNurbs/gsBSpline.h#L195) method in the gsBSpline class for my projects. However, I noticed the absence of `segmentFromTo` or `splitAt` methods, which are available in the gsTensorBSpline class. Consequently, I found myself converting `gsBSpline` curves into `gsTensorBSpline` to leverage the [splitAt](https://github.com/gismo/gismo/blob/a979dacec00b5fafb9ccea03e4a75b2682085982/src/gsNurbs/gsTensorBSpline.hpp#L449) method, followed by reverting them back to use `merge`—a process that was clearly inefficient. To address this, I introduced `segmentFromTo` and `splitAt` methods directly to the gsBSpline class, along with a new example, `bSplineCurveTrim_example.cpp`, to showcase basic trim and merge operations on BSpline curves. I hope this may benefit other users of gismo.

NEW:
`segmentFromTo` and `splitAt` methods to gsBSpline class;
`bSplineCurveTrim_example.cpp` to demonstrate the basic trim/merge operations on BSpline curves

IMPROVED: no

FIXED: no

API: no

 ---------

Please consider the following checklist before issuing a pull
request:
- [X] Have you added an explanation of what your changes do and why
  you'd like us to include them?
- [X] Have you documented any new codes using Doxygen comments?
- [X] Have you written new tests or examples for your changes?
-----

jiyess committed:
add segmentFromTo and splitAt methods to gsBSpline class, and add a bSplineCurveTrim_example.cpp to demonstrate the basic trim/merge operations on BSpline curves

filiatra reviewed (changes requested):
Hi Ye,
thnx for the additions.

The content of your example could be added in the existing example file
bSplineCurve_example.cpp
so that keep fewer files in the examples folder. Please also add your name in the author list in that example.

jiyess committed:
remove bSplineCurveTrim_example.cpp, and example added into the existing exampl file bSplineCurve_example.cpp

jiyess commented:
Hi Angelos,

I have deleted the `bSplineCurveTrim_example.cpp`, and added this example into the existing example file `bSplineCurve_example.cpp`.

Best, 
Ye

jiyess committed:
test ambiguous gsBSpline::insertKnot and gsGeometry::insertKnot, we leave one copy.insertKnot and another one copy.gsBSpline<T>::insertKnot

jiyess committed:
fix ambiguous function problem of gsBSpline::insertKnot and gsGeometry::insertKnot by calling copy.gsBSpline<T>::insertKnot

filiatra committed:
Merge remote-tracking branch 'origin/stable' into trimCruve_for_gsBSpline

filiatra committed:
fix ambigius functions
